server/http.go

```diff
@@ -84,6 +92,18 @@ func parseRequestURL(url string) (cache.EntryKind, string, error) {
 func (h *httpCache) CacheHandler(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
 
+	// Exit bazel-remote when there is no request for MaxIdleTime seconds
+	if MaxIdleTime > 0 {
+		if idleTimer != nil {
+			idleTimer.Stop()
+		}
+		idleTimer := time.AfterFunc(time.Duration(MaxIdleTime)*time.Second, func() {
+			h.accessLogger.Printf("Exit bazel-remote server after no request for %d seconds", MaxIdleTime)
+			os.Exit(0)
```

Comment:
One issue with calling `os.Exit()` is that it doesn't properly terminate open network connections (bazel-remote uses http keep-alive) and another problem with `os.Exit()` is that it's difficult to test (I think).

An alternative might be to use the https://godoc.org/net/http#Server.Shutdown function (see the example in godoc). The timer could then be implemented in `main.go` to periodically (once a second) check whether the last request has been served longer than `idle_timeout` seconds ago, if so it calls server. In order to do so simply add a lastRequestTime() function to the HTTPCache interface.

What do you think?

Reply:
updated, thanks!